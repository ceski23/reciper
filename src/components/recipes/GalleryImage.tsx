/* eslint-disable @typescript-eslint/no-use-before-define */
import styled from '@emotion/styled';
import { FC } from 'react';
import { Item } from 'react-photoswipe-gallery';

import { useImageDimensions } from 'hooks/useImageDimensions';

interface GalleryImageProps {
  url: string;
}

export const GalleryImage: FC<GalleryImageProps> = ({ url }) => {
  const dimensions = useImageDimensions(url);

  return (
    <Item
      original={url}
      thumbnail={url}
      key={url}
      width={dimensions?.width}
      height={dimensions?.height}
      cropped
    >
      {({ ref, open }) => (
        <ImageThumbnail
          ref={ref as React.MutableRefObject<HTMLImageElement>}
          onClick={open}
          src={url}
        />
      )}
    </Item>
  );
};

const ImageThumbnail = styled.img`
  width: 100%;
  height: auto;
  aspect-ratio: 1;
  object-fit: cover;
  border-radius: 10px;
  cursor: pointer;
`;
