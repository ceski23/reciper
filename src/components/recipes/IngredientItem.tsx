import { css, Theme } from '@emotion/react';
import styled from '@emotion/styled';
import { VFC } from 'react';

import { ReactComponent as GroceryIcon } from 'assets/recipes/grocery.svg';

import { useAppSelector } from 'hooks/store';

import ingredientsDatabase from 'services/ingredients/database';
import { ParsedIngredient } from 'services/ingredients/models';
import { units } from 'services/units';
import { convertIngredient } from 'services/units/utils';

import { selectConversionPrecision, selectUnitsConversions, selectUseUnitsConversion } from 'store/settings';

import { color } from 'utils/styles/theme';
import { fluidTypography } from 'utils/styles/typography';

interface Props {
  ingredient: ParsedIngredient;
  onClick?: () => void;
}

const iconStyles = ({ theme }: { theme: Theme }) => css`
  border-radius: 10px;
  background-color: ${theme.colors.backgroundAlt};
  width: 50px;
  height: 50px;
  border: none;
  margin-right: 20px;
  flex-shrink: 0;
  padding: 10px;
`;

const Icon = styled.img`
  ${iconStyles}

  object-fit: contain;
`;

const FallbackIcon = styled(GroceryIcon)`
  ${iconStyles}

  fill: ${color('text')};
`;

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

const Text = styled.p`
  flex-grow: 1;
  margin: 0;
  ${({ theme }) => fluidTypography(
    theme.breakpoints.small,
    theme.breakpoints.xlarge,
    16,
    20,
  )}
`;

const formatQuantity = (quantity: number, precision?: number) => {
  if (precision === undefined) return quantity;
  return Number(quantity.toFixed(precision));
};

export const IngredientItem: VFC<Props> = ({ ingredient, onClick }) => {
  const conversions = useAppSelector(selectUnitsConversions);
  const conversionEnabled = useAppSelector(selectUseUnitsConversion);
  const convertionPrecision = useAppSelector(selectConversionPrecision);

  const precision = conversionEnabled ? convertionPrecision : undefined;

  const convertedIngredient = conversionEnabled ? convertIngredient(
    ingredient,
    ('type' in ingredient && ingredient.type) ? conversions[ingredient.type] : undefined,
  ) : ingredient;

  const quantity = ('parsed' in convertedIngredient)
    ? formatQuantity(convertedIngredient.quantity, precision)
    : undefined;

  const unitText = ('unit' in convertedIngredient)
    ? units[convertedIngredient.unit].formatter.format({
      quantity: formatQuantity(convertedIngredient.quantity, precision),
    })
    : undefined;

  return (
    <Wrapper onClick={onClick}>
      {('type' in ingredient && ingredient.type) ? (
        <Icon src={ingredientsDatabase[ingredient.type].image} />
      ) : (
        <FallbackIcon />
      )}

      <Text>
        {('parsed' in ingredient) ? (
          <>
            {ingredient.parsed.begin}{' '}
            <strong>{quantity} {unitText}</strong>
            {' '}{ingredient.parsed.end}
          </>
        ) : ingredient.original}
      </Text>
    </Wrapper>
  );
};
