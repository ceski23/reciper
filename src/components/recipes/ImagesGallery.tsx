/* eslint-disable @typescript-eslint/no-use-before-define */
import styled from '@emotion/styled';
import { FC } from 'react';
import { Gallery } from 'react-photoswipe-gallery';

import { GalleryImage } from 'components/recipes/GalleryImage';

interface ImagesGalleryProps {
  images: string[]
}

export const ImagesGallery: FC<ImagesGalleryProps> = ({ images }) => (
  <div>
    <h2>Galeria</h2>
    <Gallery>
      <ImagesContainer>
        {images.map((image) => (
          <GalleryImage key={image} url={image} />
        ))}
      </ImagesContainer>
    </Gallery>
  </div>
);

export default ImagesGallery;

const ImagesContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(100px, 1fr));
  gap: 20px;
`;
