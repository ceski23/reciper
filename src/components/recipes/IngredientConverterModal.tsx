import styled from '@emotion/styled';
import { ChangeEvent, FC } from 'react';

import { ReactComponent as ConversionIcon } from 'assets/recipes/conversion.svg';

import { Modal, ModalProps } from 'components/common/modal/Modal';
import { Field } from 'components/forms/inputs/Field';

import { useIngredientConversion } from 'hooks/recipes/useIngredientConversion';

import KNOWN_INGREDIENTS from 'services/ingredients/database';
import { IngredientWithUnitAndType } from 'services/ingredients/models';
import { UnitName, units } from 'services/units';

import { color } from 'utils/styles/theme';

const IngredientRow = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 20px;
  font-weight: 700;
  margin-bottom: 20px;
`;

const IngredientImage = styled.img`
  width: 50px;
  height: 50px;
  object-fit: contain;
`;

const StyledConversionIcon = styled(ConversionIcon)`
  width: 50px;
  height: 50px;
  fill: ${color('textalt')};
  flex-grow: 1;
  flex-shrink: 0;
`;

const ConversionRow = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 10px;
`;

const Select = styled.select`
  border: none;
  background: none;
  font-size: inherit;
  color: inherit;
`;

const ConversionContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  row-gap: 10px;
`;

const QuantityField = styled(Field)`
  max-width: 100px;
  min-width: 50px;
  text-align: end;

  &::-webkit-outer-spin-button,
  &::-webkit-inner-spin-button {
      /* display: none; <- Crashes Chrome on hover */
      -webkit-appearance: none;
      margin: 0; /* <-- Apparently some margin are still there even though it's hidden */
  }

  &[type=number] {
      -moz-appearance:textfield; /* Firefox */
  }
`;

const capitalize = (text: string) => (
  `${text[0].toLocaleUpperCase()}${text.substring(1)}`
);

interface IngredientConverterModalProps {
  ingredient: IngredientWithUnitAndType;
}

const stringToNumber = (text: string) => {
  const num = Number.parseFloat(text);
  return Number.isNaN(num) ? 0 : num;
};

export const IngredientConverterModal: FC<IngredientConverterModalProps & ModalProps> = ({
  ingredient, ...props
}) => {
  const availableUnits = Object.keys(KNOWN_INGREDIENTS[ingredient.type].conversions) as UnitName[];
  const ingredientObject = KNOWN_INGREDIENTS[ingredient.type];
  const {
    changeQuantity, changeUnit, conversionData, swapPositions,
  } = useIngredientConversion(ingredient);

  const handleQuantityChange = (event: ChangeEvent<HTMLInputElement>) => {
    const fieldName = event.currentTarget.name;
    const quantity = stringToNumber(event.currentTarget.value);

    if (fieldName === 'from') changeQuantity({ from: quantity });
    else if (fieldName === 'to') changeQuantity({ to: quantity });
  };

  const handleUnitChange = (event: ChangeEvent<HTMLSelectElement>) => {
    const fieldName = event.currentTarget.name;
    const unit = event.currentTarget.value as UnitName;

    if (fieldName === 'from') changeUnit({ from: unit });
    else if (fieldName === 'to') changeUnit({ to: unit });
  };

  return (
    <Modal {...props}>
      <Modal.Header title="Kalkulator jednostek" />
      <Modal.Body>
        <IngredientRow>
          <IngredientImage src={ingredientObject.image} alt={ingredientObject.name} />
          <p>{capitalize(ingredientObject.name)}</p>
        </IngredientRow>

        <ConversionContainer>
          <ConversionRow>
            <QuantityField
              id="fromQuantity"
              type="number"
              name="from"
              inputMode="decimal"
              value={conversionData.quantity.from}
              onChange={handleQuantityChange}
              min={0}
            />

            <Select value={conversionData.unit.from} onChange={handleUnitChange} name="from">
              {availableUnits.map((unit) => (
                <option key={unit} value={unit}>
                  {units[unit].formatter.format({ quantity: conversionData.quantity.from })}
                </option>
              ))}
            </Select>
          </ConversionRow>

          <StyledConversionIcon onClick={swapPositions} />

          <ConversionRow>
            <QuantityField
              id="fromQuantity"
              type="number"
              name="to"
              inputMode="decimal"
              value={conversionData.quantity.to}
              onChange={handleQuantityChange}
              min={0}
            />

            <Select value={conversionData.unit.to} onChange={handleUnitChange} name="to">
              {availableUnits.map((unit) => (
                <option key={unit} value={unit}>
                  {units[unit].formatter.format({ quantity: conversionData.quantity.to })}
                </option>
              ))}
            </Select>
          </ConversionRow>
        </ConversionContainer>
      </Modal.Body>
      <Modal.Footer cancelText="Zamknij" />
    </Modal>
  );
};
