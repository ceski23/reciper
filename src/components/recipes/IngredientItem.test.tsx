import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { renderWithProviders } from 'test/utils';

import { IngredientItem } from 'components/recipes/IngredientItem';

import { ParsedIngredient } from 'services/ingredients/models';
import { parseIngredient } from 'services/ingredients/parser';

describe('IngredientItem', () => {
  it('should be clickable', async () => {
    const ingredient: ParsedIngredient = parseIngredient('szczypta magii');
    const handleClick = vi.fn();
    const user = userEvent.setup();

    renderWithProviders(
      <IngredientItem ingredient={ingredient} onClick={handleClick} />,
    );

    const element = screen.getByText(/szczypta magii/i);
    await user.click(element);
    expect(handleClick).toBeCalledTimes(1);
  });

  it('should render with UnknownIngredient', () => {
    const ingredient: ParsedIngredient = parseIngredient('szczypta magii');

    const { asFragment } = renderWithProviders(
      <IngredientItem ingredient={ingredient} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with IngredientWithQuantity', () => {
    const ingredient: ParsedIngredient = parseIngredient('duże 3 wiaderka betonu');

    const { asFragment } = renderWithProviders(
      <IngredientItem ingredient={ingredient} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with IngredientWithUnit', () => {
    const ingredient: ParsedIngredient = parseIngredient('3 kilogramy betonu');

    const { asFragment } = renderWithProviders(
      <IngredientItem ingredient={ingredient} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with IngredientWithQuantityAndType', () => {
    const ingredient: ParsedIngredient = parseIngredient('3 wiadra masła');

    const { asFragment } = renderWithProviders(
      <IngredientItem ingredient={ingredient} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with IngredientWithUnitAndType', () => {
    const ingredient: ParsedIngredient = parseIngredient('3 kilogramy masła');

    const { asFragment } = renderWithProviders(
      <IngredientItem ingredient={ingredient} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
