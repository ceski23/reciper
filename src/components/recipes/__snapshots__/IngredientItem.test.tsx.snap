// Vitest Snapshot v1

exports[`IngredientItem > should render with IngredientWithQuantity 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  border-radius: 10px;
  background-color: #dfe8ec;
  width: 50px;
  height: 50px;
  border: none;
  margin-right: 20px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 10px;
  fill: #023;
}

.emotion-4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  font-size: 16px;
  font-size: clamp(16px, 12.307692307692308px + 0.641025641025641vw, 20px);
}

<div
    class="emotion-0 emotion-1"
  >
    <svg
      class="emotion-2 emotion-3"
      fill="currentColor"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M496 184h-8V95.21a79.517 79.517 0 0 0-48.24-73.16 40.206 40.206 0 0 0-31.52 0A79.517 79.517 0 0 0 360 95.21v75.6a87.544 87.544 0 0 1-15.64-42.82v-.008A7.994 7.994 0 0 0 352 120V72a8 8 0 0 0-8-8h-64a8 8 0 0 0-8 8v2.85c-.44-.51-.9-1-1.38-1.48a31.984 31.984 0 0 0-50.61 7.14 32.22 32.22 0 0 0-5.39-7.14 32.003 32.003 0 0 0-53.81 15.44A31.988 31.988 0 0 0 146.86 144a31.965 31.965 0 0 0 5.14 51.7V240h16v-40h240v280h-64v16h152a7.998 7.998 0 0 0 8-8V192a7.998 7.998 0 0 0-8-8zm-120-48h24a8 8 0 0 1 0 16h-24zm4.37-64H400a8 8 0 1 1 0 16h-23.6a64.012 64.012 0 0 1 3.97-16zM288 80h48v32h-48zM168 184a16 16 0 0 1 0-32 8 8 0 0 0 0-16 16 16 0 0 1 0-32 7.998 7.998 0 0 0 8-8 16 16 0 0 1 31.62-3.47 31.958 31.958 0 0 0-4.76 51.47 31.937 31.937 0 0 0-6.56 40zm56 0a16 16 0 0 1 0-32 8 8 0 0 0 0-16 16 16 0 0 1 0-32 7.998 7.998 0 0 0 8-8 16 16 0 1 1 32 0 7.998 7.998 0 0 0 8 8v16a7.994 7.994 0 0 0 7.64 7.982v.008A87.734 87.734 0 0 1 256 180.62V96h-16v88zm50.14 0a103.386 103.386 0 0 0 21.55-56h32.62a103.386 103.386 0 0 0 21.55 56zM376 184v-16h24a24 24 0 0 0 0-48h-24v-16h24a24 24 0 0 0 0-48h-10.48a63.465 63.465 0 0 1 25.03-19.24 24.042 24.042 0 0 1 18.9 0A63.516 63.516 0 0 1 472 95.21V184zm112 296h-64V200h64zM176 432a24.043 24.043 0 0 0-18.334-23.324 67.082 67.082 0 0 0-123.332 0 24.003 24.003 0 0 0 0 46.648 67.082 67.082 0 0 0 123.332 0A24.043 24.043 0 0 0 176 432zm-24 8a8 8 0 0 0-7.428 5.03l-1.144 2.86a51.082 51.082 0 0 1-94.856 0l-1.144-2.86A8 8 0 0 0 40 440a8 8 0 0 1 0-16 8 8 0 0 0 7.428-5.03l1.144-2.86a51.082 51.082 0 0 1 94.856 0l1.144 2.86A8 8 0 0 0 152 424a8 8 0 0 1 0 16z"
      />
      <path
        d="M64 408h16v16H64zM80 440h16v16H80zM112 424h16v16h-16zM206.02 341.85l17.14-34.27-14.32-7.16-11.03 22.07-5.81-5.81v-16.93a40.185 40.185 0 0 0-34.34-39.59l-28.53-4.08a7.994 7.994 0 0 0-9.05 9.05l4.08 28.53A40.185 40.185 0 0 0 163.75 328h16.93l9.21 9.21a80.08 80.08 0 0 0-67.23 19.16l10.68 11.92A63.911 63.911 0 0 1 176 352a62.442 62.442 0 0 1 6.87.37l-9.07 18.13a7.74 7.74 0 0 0-2.22-1.66l-16-8-7.16 14.32 13.85 6.92a25.955 25.955 0 0 0 29.34 10.67l12.81 6.41 7.16-14.32-16-8a8.044 8.044 0 0 0-7.02-.07l10.28-20.55a64.007 64.007 0 0 1-44.17 120.14l-5.34 15.09a80.01 80.01 0 0 0 56.69-149.6zM176 300.69l-15.6-15.6-11.31 11.31 15.6 15.6h-.94a24.127 24.127 0 0 1-23.76-20.61l-2.56-17.96 17.96 2.57A24.109 24.109 0 0 1 176 299.75z"
      />
      <path
        d="M304 272h-8v-8h8a7.998 7.998 0 0 0 8-8v-32a7.998 7.998 0 0 0-8-8h-80a7.998 7.998 0 0 0-8 8v32a7.998 7.998 0 0 0 8 8h8v8h-8a32.051 32.051 0 0 0-23.85 10.67l11.92 10.66A16.03 16.03 0 0 1 224 288h16a7.998 7.998 0 0 0 8-8v-16h32v8h-16v16h40a16.021 16.021 0 0 1 16 16v16h-88v16h88v96h-48v16h48v16a16.021 16.021 0 0 1-16 16h-64v16h64a32.036 32.036 0 0 0 32-32V304a32.036 32.036 0 0 0-32-32zm-72-24v-16h64v16z"
      />
    </svg>
    <p
      class="emotion-4 emotion-5"
    >
      duże 
      <strong>
        3 
      </strong>
       wiaderka betonu
    </p>
  </div>
</DocumentFragment>
`;

exports[`IngredientItem > should render with IngredientWithQuantityAndType 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  border-radius: 10px;
  background-color: #dfe8ec;
  width: 50px;
  height: 50px;
  border: none;
  margin-right: 20px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 10px;
  object-fit: contain;
}

.emotion-4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  font-size: 16px;
  font-size: clamp(16px, 12.307692307692308px + 0.641025641025641vw, 20px);
}

<div
    class="emotion-0 emotion-1"
  >
    <img
      class="emotion-2 emotion-3"
      src="/src/assets/ingredients/butter.webp"
    />
    <p
      class="emotion-4 emotion-5"
    >
       
      <strong>
        3 
      </strong>
       wiadra masła
    </p>
  </div>
</DocumentFragment>
`;

exports[`IngredientItem > should render with IngredientWithUnit 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  border-radius: 10px;
  background-color: #dfe8ec;
  width: 50px;
  height: 50px;
  border: none;
  margin-right: 20px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 10px;
  fill: #023;
}

.emotion-4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  font-size: 16px;
  font-size: clamp(16px, 12.307692307692308px + 0.641025641025641vw, 20px);
}

<div
    class="emotion-0 emotion-1"
  >
    <svg
      class="emotion-2 emotion-3"
      fill="currentColor"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M496 184h-8V95.21a79.517 79.517 0 0 0-48.24-73.16 40.206 40.206 0 0 0-31.52 0A79.517 79.517 0 0 0 360 95.21v75.6a87.544 87.544 0 0 1-15.64-42.82v-.008A7.994 7.994 0 0 0 352 120V72a8 8 0 0 0-8-8h-64a8 8 0 0 0-8 8v2.85c-.44-.51-.9-1-1.38-1.48a31.984 31.984 0 0 0-50.61 7.14 32.22 32.22 0 0 0-5.39-7.14 32.003 32.003 0 0 0-53.81 15.44A31.988 31.988 0 0 0 146.86 144a31.965 31.965 0 0 0 5.14 51.7V240h16v-40h240v280h-64v16h152a7.998 7.998 0 0 0 8-8V192a7.998 7.998 0 0 0-8-8zm-120-48h24a8 8 0 0 1 0 16h-24zm4.37-64H400a8 8 0 1 1 0 16h-23.6a64.012 64.012 0 0 1 3.97-16zM288 80h48v32h-48zM168 184a16 16 0 0 1 0-32 8 8 0 0 0 0-16 16 16 0 0 1 0-32 7.998 7.998 0 0 0 8-8 16 16 0 0 1 31.62-3.47 31.958 31.958 0 0 0-4.76 51.47 31.937 31.937 0 0 0-6.56 40zm56 0a16 16 0 0 1 0-32 8 8 0 0 0 0-16 16 16 0 0 1 0-32 7.998 7.998 0 0 0 8-8 16 16 0 1 1 32 0 7.998 7.998 0 0 0 8 8v16a7.994 7.994 0 0 0 7.64 7.982v.008A87.734 87.734 0 0 1 256 180.62V96h-16v88zm50.14 0a103.386 103.386 0 0 0 21.55-56h32.62a103.386 103.386 0 0 0 21.55 56zM376 184v-16h24a24 24 0 0 0 0-48h-24v-16h24a24 24 0 0 0 0-48h-10.48a63.465 63.465 0 0 1 25.03-19.24 24.042 24.042 0 0 1 18.9 0A63.516 63.516 0 0 1 472 95.21V184zm112 296h-64V200h64zM176 432a24.043 24.043 0 0 0-18.334-23.324 67.082 67.082 0 0 0-123.332 0 24.003 24.003 0 0 0 0 46.648 67.082 67.082 0 0 0 123.332 0A24.043 24.043 0 0 0 176 432zm-24 8a8 8 0 0 0-7.428 5.03l-1.144 2.86a51.082 51.082 0 0 1-94.856 0l-1.144-2.86A8 8 0 0 0 40 440a8 8 0 0 1 0-16 8 8 0 0 0 7.428-5.03l1.144-2.86a51.082 51.082 0 0 1 94.856 0l1.144 2.86A8 8 0 0 0 152 424a8 8 0 0 1 0 16z"
      />
      <path
        d="M64 408h16v16H64zM80 440h16v16H80zM112 424h16v16h-16zM206.02 341.85l17.14-34.27-14.32-7.16-11.03 22.07-5.81-5.81v-16.93a40.185 40.185 0 0 0-34.34-39.59l-28.53-4.08a7.994 7.994 0 0 0-9.05 9.05l4.08 28.53A40.185 40.185 0 0 0 163.75 328h16.93l9.21 9.21a80.08 80.08 0 0 0-67.23 19.16l10.68 11.92A63.911 63.911 0 0 1 176 352a62.442 62.442 0 0 1 6.87.37l-9.07 18.13a7.74 7.74 0 0 0-2.22-1.66l-16-8-7.16 14.32 13.85 6.92a25.955 25.955 0 0 0 29.34 10.67l12.81 6.41 7.16-14.32-16-8a8.044 8.044 0 0 0-7.02-.07l10.28-20.55a64.007 64.007 0 0 1-44.17 120.14l-5.34 15.09a80.01 80.01 0 0 0 56.69-149.6zM176 300.69l-15.6-15.6-11.31 11.31 15.6 15.6h-.94a24.127 24.127 0 0 1-23.76-20.61l-2.56-17.96 17.96 2.57A24.109 24.109 0 0 1 176 299.75z"
      />
      <path
        d="M304 272h-8v-8h8a7.998 7.998 0 0 0 8-8v-32a7.998 7.998 0 0 0-8-8h-80a7.998 7.998 0 0 0-8 8v32a7.998 7.998 0 0 0 8 8h8v8h-8a32.051 32.051 0 0 0-23.85 10.67l11.92 10.66A16.03 16.03 0 0 1 224 288h16a7.998 7.998 0 0 0 8-8v-16h32v8h-16v16h40a16.021 16.021 0 0 1 16 16v16h-88v16h88v96h-48v16h48v16a16.021 16.021 0 0 1-16 16h-64v16h64a32.036 32.036 0 0 0 32-32V304a32.036 32.036 0 0 0-32-32zm-72-24v-16h64v16z"
      />
    </svg>
    <p
      class="emotion-4 emotion-5"
    >
       
      <strong>
        3 
      kilogramy
    
      </strong>
       betonu
    </p>
  </div>
</DocumentFragment>
`;

exports[`IngredientItem > should render with IngredientWithUnitAndType 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  border-radius: 10px;
  background-color: #dfe8ec;
  width: 50px;
  height: 50px;
  border: none;
  margin-right: 20px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 10px;
  object-fit: contain;
}

.emotion-4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  font-size: 16px;
  font-size: clamp(16px, 12.307692307692308px + 0.641025641025641vw, 20px);
}

<div
    class="emotion-0 emotion-1"
  >
    <img
      class="emotion-2 emotion-3"
      src="/src/assets/ingredients/butter.webp"
    />
    <p
      class="emotion-4 emotion-5"
    >
       
      <strong>
        3 
      kilogramy
    
      </strong>
       masła
    </p>
  </div>
</DocumentFragment>
`;

exports[`IngredientItem > should render with UnknownIngredient 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  border-radius: 10px;
  background-color: #dfe8ec;
  width: 50px;
  height: 50px;
  border: none;
  margin-right: 20px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 10px;
  fill: #023;
}

.emotion-4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  font-size: 16px;
  font-size: clamp(16px, 12.307692307692308px + 0.641025641025641vw, 20px);
}

<div
    class="emotion-0 emotion-1"
  >
    <svg
      class="emotion-2 emotion-3"
      fill="currentColor"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M496 184h-8V95.21a79.517 79.517 0 0 0-48.24-73.16 40.206 40.206 0 0 0-31.52 0A79.517 79.517 0 0 0 360 95.21v75.6a87.544 87.544 0 0 1-15.64-42.82v-.008A7.994 7.994 0 0 0 352 120V72a8 8 0 0 0-8-8h-64a8 8 0 0 0-8 8v2.85c-.44-.51-.9-1-1.38-1.48a31.984 31.984 0 0 0-50.61 7.14 32.22 32.22 0 0 0-5.39-7.14 32.003 32.003 0 0 0-53.81 15.44A31.988 31.988 0 0 0 146.86 144a31.965 31.965 0 0 0 5.14 51.7V240h16v-40h240v280h-64v16h152a7.998 7.998 0 0 0 8-8V192a7.998 7.998 0 0 0-8-8zm-120-48h24a8 8 0 0 1 0 16h-24zm4.37-64H400a8 8 0 1 1 0 16h-23.6a64.012 64.012 0 0 1 3.97-16zM288 80h48v32h-48zM168 184a16 16 0 0 1 0-32 8 8 0 0 0 0-16 16 16 0 0 1 0-32 7.998 7.998 0 0 0 8-8 16 16 0 0 1 31.62-3.47 31.958 31.958 0 0 0-4.76 51.47 31.937 31.937 0 0 0-6.56 40zm56 0a16 16 0 0 1 0-32 8 8 0 0 0 0-16 16 16 0 0 1 0-32 7.998 7.998 0 0 0 8-8 16 16 0 1 1 32 0 7.998 7.998 0 0 0 8 8v16a7.994 7.994 0 0 0 7.64 7.982v.008A87.734 87.734 0 0 1 256 180.62V96h-16v88zm50.14 0a103.386 103.386 0 0 0 21.55-56h32.62a103.386 103.386 0 0 0 21.55 56zM376 184v-16h24a24 24 0 0 0 0-48h-24v-16h24a24 24 0 0 0 0-48h-10.48a63.465 63.465 0 0 1 25.03-19.24 24.042 24.042 0 0 1 18.9 0A63.516 63.516 0 0 1 472 95.21V184zm112 296h-64V200h64zM176 432a24.043 24.043 0 0 0-18.334-23.324 67.082 67.082 0 0 0-123.332 0 24.003 24.003 0 0 0 0 46.648 67.082 67.082 0 0 0 123.332 0A24.043 24.043 0 0 0 176 432zm-24 8a8 8 0 0 0-7.428 5.03l-1.144 2.86a51.082 51.082 0 0 1-94.856 0l-1.144-2.86A8 8 0 0 0 40 440a8 8 0 0 1 0-16 8 8 0 0 0 7.428-5.03l1.144-2.86a51.082 51.082 0 0 1 94.856 0l1.144 2.86A8 8 0 0 0 152 424a8 8 0 0 1 0 16z"
      />
      <path
        d="M64 408h16v16H64zM80 440h16v16H80zM112 424h16v16h-16zM206.02 341.85l17.14-34.27-14.32-7.16-11.03 22.07-5.81-5.81v-16.93a40.185 40.185 0 0 0-34.34-39.59l-28.53-4.08a7.994 7.994 0 0 0-9.05 9.05l4.08 28.53A40.185 40.185 0 0 0 163.75 328h16.93l9.21 9.21a80.08 80.08 0 0 0-67.23 19.16l10.68 11.92A63.911 63.911 0 0 1 176 352a62.442 62.442 0 0 1 6.87.37l-9.07 18.13a7.74 7.74 0 0 0-2.22-1.66l-16-8-7.16 14.32 13.85 6.92a25.955 25.955 0 0 0 29.34 10.67l12.81 6.41 7.16-14.32-16-8a8.044 8.044 0 0 0-7.02-.07l10.28-20.55a64.007 64.007 0 0 1-44.17 120.14l-5.34 15.09a80.01 80.01 0 0 0 56.69-149.6zM176 300.69l-15.6-15.6-11.31 11.31 15.6 15.6h-.94a24.127 24.127 0 0 1-23.76-20.61l-2.56-17.96 17.96 2.57A24.109 24.109 0 0 1 176 299.75z"
      />
      <path
        d="M304 272h-8v-8h8a7.998 7.998 0 0 0 8-8v-32a7.998 7.998 0 0 0-8-8h-80a7.998 7.998 0 0 0-8 8v32a7.998 7.998 0 0 0 8 8h8v8h-8a32.051 32.051 0 0 0-23.85 10.67l11.92 10.66A16.03 16.03 0 0 1 224 288h16a7.998 7.998 0 0 0 8-8v-16h32v8h-16v16h40a16.021 16.021 0 0 1 16 16v16h-88v16h88v96h-48v16h48v16a16.021 16.021 0 0 1-16 16h-64v16h64a32.036 32.036 0 0 0 32-32V304a32.036 32.036 0 0 0-32-32zm-72-24v-16h64v16z"
      />
    </svg>
    <p
      class="emotion-4 emotion-5"
    >
      szczypta magii
    </p>
  </div>
</DocumentFragment>
`;
